### PR TITLE
Add Poppler

### PR DESCRIPTION
**Pull request by ericphanson**
This doesn't work yet, giving the following error:

```julia

CMake Error at /opt/i686-linux-gnu/i686-linux-gnu/sys-root/usr/local/lib/openjpeg-2.3/OpenJPEGTargets.cmake:78 (message):
  The imported target "opj_decompress" references the file

     "/opt/i686-linux-gnu/i686-linux-gnu/sys-root/usr/local/bin/opj_decompress"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/opt/i686-linux-gnu/i686-linux-gnu/sys-root/usr/local/lib/openjpeg-2.3/OpenJPEGTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /opt/i686-linux-gnu/i686-linux-gnu/sys-root/usr/local/lib/openjpeg-2.3/OpenJPEGConfig.cmake:28 (include)
  CMakeLists.txt:205 (find_package)


-- Configuring incomplete, errors occurred!
See also "/workspace/srcdir/poppler-0.87.0/CMakeFiles/CMakeOutput.log".
See also "/workspace/srcdir/poppler-0.87.0/CMakeFiles/CMakeError.log".
Previous command exited with 1
ERROR: LoadError: Build for Poppler on i686-linux-gnu did not complete successfully
```

but I'm putting it up in hopes of getting some help figuring out what's wrong.

I've commented out `GTK3` and `gdk_pixbuf` since I've set `DENABLE_GLIB=OFF` to start easy, but we might want that at some point.

### CODE REVIEW

**Comment by giordano** [comment]
> I've commented out `GTK3` and `gdk_pixbuf` since I've set `DENABLE_GLIB=OFF` to start easy, but we might want that at some point.

Does `ENABLE_GLIB` mean "require GTK3" and not simply "require Glib"?

**Comment by ericphanson** [comment]
LFS says GTK+-3.24.18 (http://www.linuxfromscratch.org/blfs/view/svn/general/poppler.html) while homebrew  says glib (https://formulae.brew.sh/formula/poppler)

**Comment by giordano** [comment]
Also Debian doesn't seem to require GTK: https://packages.debian.org/source/sid/poppler

File: P/Poppler/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+platforms = supported_platforms()
+
+
+# The products that we will ensure are always built
+products = Product[
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("JpegTurbo_jll"),
```
Comment (giordano) [inline P/Poppler/build_tarballs.jl:39]: Add `Xorg_xorgproto_jll` as build-dependency:
```suggestion
    BuildDependency("Xorg_xorgproto_jll.jl"),
    Dependency("JpegTurbo_jll"),
```

**Comment by giordano** [comment]
I removed `poppler-data`, it's just extra data not needed for compilation.  Why did you add it?

**Comment by ericphanson** [comment]
I wasn't sure if I should or not; it was described as "encoding data" which made me think maybe there'd be encoding issues if it was missing. And since encoding issues are awful I thought I'd be on the safe side by including it.

**Comment by ericphanson** [comment]
Thanks very much for the fixes! Especially the OpenJpeg stuff, I found the errors there very confusing.

**Comment by giordano** [comment]
> I wasn't sure if I should or not; it was described as "encoding data" which made me think maybe there'd be encoding issues if it was missing. And since encoding issues are awful I thought I'd be on the safe side by including it.

If it's needed at runtime, my suggestion is to add it in an `Artifacts.toml` in your package.  In the recipe you were even installing it after poppler

**Comment by ericphanson** [comment]
Thanks very much for all the help @giordano!

**Comment by mirkobunse** [comment]
Hi guys!

Me and the maintainers from `JuliaTeX` have lately discussed the loading times from packages that depend on `Poppler_jll` (see the referenced issues above). To be specific, these packages do not require all functionality of poppler, only `pdftocairo`.

Apparently, `@time using Poppler_jll` is not very fast (2.3 seconds on julia-1.5, almost 7 seconds on julia-1.3) and it seems to slow down the dependent packages. While we already know that loading times of `*_jll` packages is an issue of its own right and things have improved a lot since julia-1.5, we have wondered:

Would a clone of your `build_tarballs.jl`, which just exposed `pdftocairo`, be loaded faster?

We are no experts in the Yggdrasil universe, so we'd deeply appreciate any comments on this idea.

**Comment by giordano** [comment]
No, the number of products wouldn't affect loading time that much.

Note that JLL packages will get a huge overhaul for julia v1.6 and loading time should be drastically reduced: it should hopefully go down to less of a second for most packages, if not all

**Pull request merged**